### PR TITLE
[fix](cloud) fix BlockFileCache::get_or_set crash

### PR DESCRIPTION
**Pull request by freemandealer**
### What problem does this PR solve?

This is a Plan-B fix, the root cause is under study:

 2# JVM_handle_linux_signal in /usr/lib/jvm/java-17-openjdk-amd64/lib/server/libjvm.so
 3# 0x00007F8CFE720520 in /lib/x86_64-linux-gnu/libc.so.6
 4# std::_Rb_tree_decrement(std::_Rb_tree_node_base*) at ../../../../../libstdc++-v3/src/c++98/tree.cc:123
 5# doris::io::BlockFileCache::get_impl[abi:cxx11](doris::io::UInt128Wrapper const&, doris::io::CacheContext const&, doris::io::FileBlock::Range const&, std::lock_guard<std::mutex>&) at /home/zcp/repo_center/doris_branch-3.0/doris/be/src/io/cache/block_file_cache.cpp:365
 6# doris::io::BlockFileCache::get_or_set(doris::io::UInt128Wrapper const&, unsigned long, unsigned long, doris::io::CacheContext&) at /home/zcp/repo_center/doris_branch-3.0/doris/be/src/io/cache/block_file_cache.cpp:662
 7# doris::io::CachedRemoteFileReader::read_at_impl(unsigned long, doris::Slice, unsigned long*, doris::io::IOContext const*) at /home/zcp/repo_center/doris_branch-3.0/doris/be/src/io/cache/cached_remote_file_reader.cpp:179

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by freemandealer** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/io/cache/block_file_cache.cpp [hunk 1]
```diff
@@ -393,6 +393,9 @@ FileBlocks BlockFileCache::get_impl(const UInt128Wrapper& hash, const CacheConte
 
     auto& file_blocks = it->second;
     DCHECK(!file_blocks.empty());
+    if (file_blocks.empty()) {
```
Comment (gavinchou) [inline be/src/io/cache/block_file_cache.cpp:396]: should we remove the unexpected key?
and we have need to print some waring for this exceptional branch, e.g., hex(key), path and etc.

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by freemandealer** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**